.github/workflows/actions.yml

```diff
@@ -127,3 +129,71 @@ jobs:
         grunt test
       env:
         CI: true
+  publish:
+    name: Publish to PyPI registry
+    needs:
+      - build_python
+      - build_javascript
+    runs-on: ubuntu-latest
+
+    env:
+      PY_COLORS: 1
+      TOXENV: packaging
+
+    steps:
+      - name: Switch to using Python 3.6 by default
+        uses: actions/setup-python@v2
+        with:
+          python-version: 3.6
+      - name: Install tox
+        run: python -m pip install --user tox
+      - name: Check out src from Git
+        uses: actions/checkout@v2
+        with:
+          # Get shallow Git history (default) for tag creation events
+          # but have a complete clone for any other workflows.
+          # Both options fetch tags but since we're going to remove
+          # one from HEAD in non-create-tag workflows, we need full
+          # history for them.
+          fetch-depth: >-
+            ${{
+              (
+                github.event_name == 'create' &&
+                github.event.ref_type == 'tag'
+              ) &&
+              1 || 0
+            }}
+      - name: Drop Git tags from HEAD for non-tag-create events
```

Comment:
just out of curiosity, why do we want to drop all tags for non tag create events?

Reply:
I do not remember exactly as I was not the original author of this task, maybe @webknjaz could help here?

My guess is to avoid getting setuptools-scm confused about which tag it would use for producing the version, especially as it is possible to add a tag somewhere back-in-time (before more recent tag in the branch).

Reply:
@ssbarnea yes, that would be one subcase.
Although, I've been doing this in particular to avoid a slightly different race condition.
When the tags are pushed, this generates 2 workflows: for tag `push` and for tag `create`.
This means that whenever both of these workflows will run at about the same time and the checkout action will get the tag and the commits. And then, setuptools-scm will generate a version with that tag exactly. And this is fine because later in the workflow, the publish step is executed under only one of these cases so both jobs won't attempt to upload. That's why this step is skipped for these cases.
Now, there's a third job that may run separately in the context of the same commit: `push` to master. And if that commit is also tagged (for example, if the tagging happens fast after pushing the same commit to master, before some of the jobs did checkout), setuptools-scm will generate a version based on that tag. This is bad when that "Publish to test.pypi.org" step is uncommented because it'll result in two jobs attempting to publish an artifact with the same name to TestPyPI and one of them will fail (with no guarantee which one). And so for this case, I'm removing any tags that may be present in the HEAD which eliminates the race condition.

Reply:
Thank you for clarifying @webknjaz. Since @nicoddemus also had the same question, I'll ask him to resolve this if this answer works for him

Reply:
Thanks @webknjaz for the explanation.

Under normal circumstances in our release workflow the race condition cannot really happen, because the tag will be created in the last commit of the release PR, and only after that is green (and hence the package is published), the commit will be merged to `master`, not as fast-forward but as a normal merge commit.

So I'm under the opinion that this is a bit of complication that really doesn't bring anything to the table for `pytest-dev/pytest-html`, but it is also harmless so feel free to go ahead with it. 👍